### PR TITLE
[MINOR] Spelling sql not core

### PR DESCRIPTION
**Pull request by jsoref**
### What changes were proposed in this pull request?

This PR intends to fix typos in the sub-modules:
* `sql/catalyst`
* `sql/hive-thriftserver`
* `sql/hive`

Split per @srowen https://github.com/apache/spark/pull/30323#issuecomment-728981618

NOTE: The misspellings have been reported at https://github.com/jsoref/spark/commit/706a726f87a0bbf5e31467fae9015218773db85b#commitcomment-44064356

### Why are the changes needed?

Misspelled words make it harder to read / understand content.

### Does this PR introduce _any_ user-facing change?

There are various fixes to documentation, etc...

### How was this patch tested?

No testing was performed

### CODE REVIEW

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 [hunk 1]
```diff
@@ -21,7 +21,7 @@ grammar SqlBase;
    * When false, INTERSECT is given the greater precedence over the other set
    * operations (UNION, EXCEPT and MINUS) as per the SQL standard.
    */
-  public boolean legacy_setops_precedence_enbled = false;
+  public boolean legacy_setops_precedence_enabled = false;
```
Comment (srowen) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4:24]: Oh hm I wonder if this has any implications elsewhere, like does this affect any non-private generated code?

Comment (jsoref) [reply sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4:24]: GitHub says maybe: https://github.com/search?q=legacy_setops_precedence_enbled&type=code

I'm going to drop this for now, we can revisit it separately.

**Comment by srowen** [comment]
Looking OK, just needs a rebase

**Review by maropu — APPROVED** [review]

**Comment by maropu** [comment]
Jenkins is down now, so could you re-invoke the GA tests?

**Comment by srowen** [comment]
Jenkins test this please

**Comment by srowen** [comment]
Jenkins test this please

**Comment by srowen** [comment]
Jenkins retest this please

**Comment by maropu** [comment]
retest this please

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**